distill/sessions/session.py

```diff
@@ -0,0 +1,270 @@
+            else:
+                tab_key = 'unknown' 
+            if tab_key not in grouped_data:
+                grouped_data[tab_key] = []
+            grouped_data[tab_key].append(key)
+        return grouped_data
+
+    def __match_url(url, pattern):
+        # Escape dots in the pattern since dot is a special character in regex
+        # and replace '*' with '.*' to match any characters sequence
+        regex_pattern = re.escape(pattern).replace('\\*', '.*')
+        
+        # Add anchors to match the entire string
+        regex_pattern = '^' + regex_pattern + '$'
+        
+        # Compile the regex pattern
+        compiled_pattern = re.compile(regex_pattern)
+        
+        # Check if the URL matches the pattern
+        return bool(compiled_pattern.match(url))
+    
+    def __flatten_dict(orig_dict, sep="_"):
+        """
+        Given a possibly nested dictionary containing logs, make a flat
+        dictionary where each key-value pair represent one user session
+        """
+        new_dict = {}
+        for first_key in orig_dict:
+            if (isinstance(orig_dict[first_key], dict)):
+                for second_key in orig_dict[first_key]:
+                    if (isinstance(orig_dict[first_key][second_key], dict)):
+                        for time_key in orig_dict[first_key][second_key]:
+                            combined_key = first_key + sep + second_key + sep + time_key
+                            new_dict[combined_key] = orig_dict[first_key][second_key][time_key]
+                    else:
+                        combined_key = first_key + sep + second_key
+                        new_dict[combined_key] = orig_dict[first_key][second_key]
+            else:
+                new_dict[first_key] = orig_dict[first_key]
+
+        return new_dict
+
+    def __chunk_by_domain(log, re):
+        """
+        Separate logs by the site that users interact with
+        :param log: Userale log in the form of dictionary
+        :param log: 
+        :return: A dictionary that represent sets separated by unique browserSessionId
+        """
+        grouped_data = {}
+        for key in log:
+            # Depending on the log types, tabID can be inside the details element
+            if 'pageUrl' in key:
+                domain = 'domain' + urlparse(key['pageUrl']).netloc
+                # Filter with the "re" parameter
+                if (re != "."):
+                    if (Session.__match_url(domain, re)):
+                        domain_key = 'domain' + re
+                    else:
+                        #Does not match, so we are skipping it
+                        continue
+                else:
+                    domain_key = domain
+            else:
+                domain_key = 'unknown' 
+
+            if domain_key not in grouped_data:
+                grouped_data[domain_key] = []
+            grouped_data[domain_key].append(key)
+        return grouped_data
```

Comment:
None of these are truly "methods" in the sense that none of them operate on self or cls. They're helpers that we should move to a `utils.py`.